lib/dor/was_seed/thumbnail_generator_service.rb

```diff
@@ -7,32 +7,29 @@ class ThumbnailGeneratorService
       # because this date is earlier than any of the archived dates of the content,
       # this tells openwayback to provide the earliest capture date.
       DATE_TO_TRIGGER_EARLIEST_CAPTURE_DATE = '19900101120000'.freeze
-      def self.capture_thumbnail(druid, workspace, uri)
-        thumbnail_file = "#{DruidTools::Druid.new(druid, workspace).content_dir}/thumbnail.jp2"
-        wayback_uri    = "#{Settings.was_seed.wayback_uri}/#{DATE_TO_TRIGGER_EARLIEST_CAPTURE_DATE}/#{uri}"
-        temporary_file = "tmp/#{druid[6, 14]}"
-        jpeg_file = "#{temporary_file}.jpeg"
+      def self.capture_thumbnail(druid, workspace, seed_uri)
+        screenshot_jpeg = "tmp/#{druid[6, 14]}.jpeg"
         begin
-          indexed?(uri)
-          capture(wayback_uri, jpeg_file)
+          indexed?(seed_uri)
+          wayback_uri = "#{Settings.was_seed.wayback_uri}/#{DATE_TO_TRIGGER_EARLIEST_CAPTURE_DATE}/#{seed_uri}"
+          screenshot(wayback_uri, screenshot_jpeg)
+          resize_jpeg(screenshot_jpeg)
+          thumbnail_file = "#{DruidTools::Druid.new(druid, workspace).content_dir}/thumbnail.jp2"
+          Assembly::Image.new(screenshot_jpeg).create_jp2(output: thumbnail_file)
+          FileUtils.rm(screenshot_jpeg, force: true)
         rescue => e
-          File.delete(jpeg_file) if File.exist?(jpeg_file)
-          raise "Thumbnail for druid #{druid} and #{uri} can't be generated.\n #{e.message}"
+          File.delete(screenshot_jpeg) if File.exist?(screenshot_jpeg)
+          raise "Thumbnail for druid #{druid} and #{seed_uri} can't be generated.\n #{e.message}"
         end
-
-        resize_temporary_image(jpeg_file)
-        Assembly::Image.new(jpeg_file).create_jp2(output: "#{temporary_file}.jp2")
-        FileUtils.rm jpeg_file
-        FileUtils.mv "#{temporary_file}.jp2", thumbnail_file
```

Comment:
this move is no longer needed - we can write the jp2 directly to the correct location;  there is nothing gained from writing it to a diff location and then moving.